tests/Jira/ApiTest.php

```diff
@@ -257,6 +257,147 @@ public function testGetPriorities()
 		$this->assertEquals($expected, $this->api->getPriorities(), 'Calling twice did not yield the same results');
 	}
 
+	public function testGetIssue()
+	{
+		$response = file_get_contents(__DIR__ . '/resources/api_get_issue.json');
+
+		$issue_key = 'POR-1';
+
+		$this->expectClientCall(
+			Api::REQUEST_GET,
+			'/rest/api/2/issue/' . $issue_key,
+			array('expand' => ''),
+			$response
+		);
+
+		$actual = $this->api->getIssue($issue_key);
+
+		$expected = json_decode($response, true);
+		$this->assertEquals($expected, $actual);
+	}
+
+	public function testEditIssue()
+	{
+		$issue_key = 'POR-1';
+		$params = array('update' => (object) array('summary' => array( (object) array("set" => "Bug in business logic"))));
```

Comment:
It does generate another JSON structure which matches the example on their REST doc page. (Arrays are encoded as [] and objects as {}) Not sure how flexible their current implementation is though but figured its best to keep it as close to the docs as possible.